File: backend/src/models/product_model.js
const mongoose = require('mongoose');

const productSchema = new mongoose.Schema({
    produto: {type: String, required: true,},
    codigo: {type: String, required: true,},
    lote: {type: String, required: true,},
    preco_unitario: {type: String, required: true,},
    quantidade: {type: String, required: true,},
    validade: {type: String, required: true,},
    categoria: {type: String, required: true,},
    marca: {type: String, required: true,},
});

const Product = mongoose.model('Product', productSchema);

module.exports = Product;

File: lib/model/product_model.dart
class Product {
  String id;
  String produto;
  String codigo;
  String lote;
  String precoUnitario;
  String quantidade;
  String validade;
  String categoria;
  String marca;

  Product({
    required this.id,
    required this.produto,
    required this.codigo,
    required this.lote,
    required this.precoUnitario,
    required this.quantidade,
    required this.validade,
    required this.categoria,
    required this.marca,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['_id'],
      produto: json['produto'],
      codigo: json['codigo'],
      lote: json['lote'],
      precoUnitario: json['preco_unitario'],
      quantidade: json['quantidade'],
      validade: json['validade'],
      categoria: json['categoria'],
      marca: json['marca'],
    );
  }
}